Frameworks/UIKit/UIViewController.mm

```diff
@@ -1250,18 +1256,29 @@ - (void)dismissViewControllerAnimated:(BOOL)animated completion:(void (^)(void))
         curController->priv->_parentViewController->priv->_modalViewController = nil;
     }
 
-    curController->priv->_parentViewController = nil;
-    curController->priv->_presentingViewController = nil;
-    curController->priv->_popoverPresentationController = nil;
-
     UIView* curView = [curController view];
 
     UIView* myView = [self view];
     [myView setHidden:FALSE];
 
     if (realPopoverPresented) {
```

Comment:
>realPopoverPresented [](start = 8, length = 20)

would be great if we have some comments here to clarify why we have different code path for dismiss